### ISSUE TITLE
Properly implement paypal API

### ISSUE THREAD

**Issue by danielchatfield** [codegl/tracegl#101]
You should not rely on the user returning to the site to issue a key as many things can happen to prevent this.

Either you should implement the paypal api such that the transaction is processed after returning to your site.
- User clicks buy
- Server sets up transaction with paypal
- Redirects user to paypal with token
- Paypal authorises payment and returns user to your site
- Your server then tells paypal to complete the payment and issues a key (preferably emails the key as well)

This means that if something happened after the user goes to paypal (browser crashed, request timeout etc.) it is not possible for the payment to have been made without the key being issued.

Or setup IPN so that when the payment completes the server will be notified and a key can be emailed.

I have bought the software but was not issued with a key.

**Comment by danielchatfield** [comment]
I don't want to have to dispute the payment with paypal but if I don't get a response then I will have no choice.

**Comment by datio** [comment]
It seems that [he's busy](https://twitter.com/patrickbrosset/status/373024116804055040) :/

**Comment by donaldej** [comment]
This is the same problem I had. You ever get a response (seeing as your post was 6 months ago)?

**Comment by danielchatfield** [comment]
Nope. Radio silence.
ᐧ

On 18 February 2014 19:24, madz92685 notifications@github.com wrote:

> This is the same problem I had. You ever get a response (seeing as your
> post was 6 months ago)?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/codegl/tracegl/issues/101#issuecomment-35422582
> .

## 

For contact info visit: www.danielchatfield.com

**Comment by donaldej** [comment]
That is comical. I have never heard of something like this. Develop what looks like a great piece of software, then disappear so nobody can actually buy it. Did you ever get a copy somehow?

**Comment by danielchatfield** [comment]
Yep (I had to buy it again and never got a refund) if you give me an email
I'll ping you over a copy.
ᐧ

On 18 February 2014 19:46, madz92685 notifications@github.com wrote:

> That is comical. I have never heard of something like this. Develop what
> looks like a great piece of software, then disappear so nobody can actually
> buy it. Did you ever get a copy?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/codegl/tracegl/issues/101#issuecomment-35425078
> .

## 

For contact info visit: www.danielchatfield.com
